### DISCUSSION TITLE
Current grammar spec?

### DISCUSSION THREAD

**Discussion by david542542** [duckdb/duckdb#4348] [General]
I'm wondering if there is something like a grammar spec for DuckDB? 

For example, something along the lines of an antlr file -- for example, https://github.com/pgcodekeeper/pgcodekeeper/blob/master/apgdiff/antlr-src/SQLParser.g4 (postgres), or the equivalent railroad diagram.

There are a few current ways that this is shown, for example:

- Railroad diagrams [here](https://duckdb.org/docs/sql/statements/select), but it doesn't seem comprehensive, for example it doesn't mention anything about `FILTER`, `LIMIT...PERCENT`, or what `expr` resolves to.
- The current flex/bison, which of course is definitive, but it also has a lot of things that aren't used such as [`VACUUM`](https://github.com/duckdb/duckdb/blob/ff7f24fd8e3128d94371827523dae85ebaf58713/third_party/libpg_query/grammar/statements/vacuum.y), and probably half a dozen others.

**Comment by mingodad** [comment]
See here https://github.com/duckdb/duckdb/discussions/3537#discussioncomment-7113750
